### PR TITLE
Add raw number/integer option to d3 number formats

### PR DESCRIPTION
**Pull request by ktmud**
### CATEGORY

- [x] Enhancement (new features, refinement)

### SUMMARY

@graceguo-supercat mentioned we don't even have an integer type in number formatting. So let's add one. Also add a "original number" type in case users want to see the raw number.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

image
image

### TEST PLAN

1. Go to any chart with number formatting (e.g. Big Number).
2. Make sure the supported formats work as expected.

### ADDITIONAL INFORMATION

- [x] Changes UI

### REVIEWERS

@graceguo-supercat @kristw @evans

### CODE REVIEW

File: superset-frontend/src/explore/controls.jsx
Comment (kristw) [inline superset-frontend/src/explore/controls.jsx:87]: How about changing the description 

`,d (12345.432 => 12,345)` to `Integer (12345.432 => 12,345)`

(optional) There are also constants in the package.

```ts
import { NumberFormats } from '@superset-ui/number-format';
NumberFormats.INTEGER
NumberFormats.SIGNED_INTEGER
```

**Review by kristw — APPROVED** [review]

**Review by graceguo-supercat — APPROVED** [review]

File: superset-frontend/src/explore/controls.jsx
Comment (ktmud) [reply superset-frontend/src/explore/controls.jsx:87]: Some languages uses "," for decimal points, so it's probably better for i18n by being more specific.

**Comment by villebro** [comment]
Not something directly related to this PR, but [durationFormatter](https://github.com/apache-superset/superset-ui/pull/209) expects parameters as a JSON object. It would be great to have support for specifying custom options for any available formatter in the registry, instead of having to hard code them in `setupFormatters.js`. So something where one can choose a formatter, and then pass a string containing options, which for d3 would be a regular d3 format string, and for duration formatter would be a string that is parsed into a JSON object.

**Comment by rusackas** [comment]
Just a heads-up, this is one of those oddities in `controls.jsx` that's fairly pervasive, and will need to be addressed in some way. In this case, probably moving it to `superset-ui` to be exported/imported from there.

**Comment by ktmud** [comment]
I agree the formatters should probably have a localized name and the `superset-ui` registries seem to be the perfect place for it.

Should we hold off this change before that refactor happens or merge it first since it's a fairly small change?

**Comment by graceguo-supercat** [comment]
I would vote for `merge it first`.  refactor to superset-ui registries seems a long term project.

**Comment by kristw** [comment]
@graceguo-supercat +1

**Pull request merged**
